Add rewrite headers to application-access getting started guide (#23732)
gravitational/teleport: docs/pages/application-access/getting-started.mdx

> Now start Teleport and point it to the application endpoint:

Reviewer: ```suggestion
Now start Teleport and point it to the application endpoint.
```

> This can be done by adding the application service to your teleport config, typically located at `/etc/teleport.yaml`.

Reviewer: ```suggestion
Enable the Application Service in your Teleport config, which by default is located at `/etc/teleport.yaml`.
```

> Add these lines to your config:

Reviewer 2: Let's show a complete config with auth token and proxy server set (to make this equivalent to the previous `teleport app start` command)

```yaml
app_service:
  enabled: "yes"
  apps:
  - name: "grafana"

>     uri: "https://grafana.example.com:3000"

Reviewer 2: Hmm, two of the reasons you might want to put teleport in front of Grafana are:

1. Your grafana instance is not publicly available and you want to enable connectivity to it
2. You don't have TLS certs for your Grafana deployment

The previous use of http and a localhost address showed both of those.

With this update, the URI looks like a publicly available, TLS-secured endpoint. Not necessarily wrong, but might leave the reader wondering what the point is.

    rewrite:
      headers:
      - "Host: grafana.teleport.example.com"
      - "Origin: https://grafana.teleport.example.com/"
```

Replace `https://grafana.example.com:3000` with the URL of your Grafana

> instance, in this case `"http://localhost:3000"`.

Reviewer 2: Don't forget to update this to match the above.

> Also update  `teleport.example.com` with the public domain of your teleport cluster. If you are a Teleport Cloud customer, use port 443 of your tenant's subdomain, e.g., `mytenant.teleport.sh:443`. Rewriting the headers is necessary for Grafana to recognize the Teleport reverse proxy.
> 
> Make sure to update the `app/name` and `app/uri` fields accordingly if you're using your own web application.

Reviewer: ```suggestion
Also replace `teleport.example.com` with the domain name and HTTPS port of your Teleport Proxy Service. If you are a Teleport Enterprise Cloud customer, use port 443 of your tenant's subdomain, e.g., `mytenant.teleport.sh:443`. Rewriting the headers is necessary for Grafana to recognize the Teleport reverse proxy.

Make sure to update the `app/name` and `app/uri` fields if you're using your own web application.
```

> Then start your Teleport cluster by running `sudo systemctl teleport start`.

Reviewer: We can use the `docs/pages/includes/start-teleport.mdx` partial for this

> To check and see that the Teleport service has started successfully you can tail the logs with `journalctl -fu teleport`.

Reviewer: ````suggestion
To check whether the Teleport service has started successfully you can tail the logs with the following command: 

```code
$ journalctl -fu teleport
```
````
So this is easier to copy/paste

Reviewer 2: Technically `systemctl status` is probably the better way _to check whether the service has started succesfully_.

The logs are a nice way to see what went wrong if it didn't start successfully.
